src/app/bl-next-linker/bl-next-linker.service.ts

```diff
@@ -0,0 +1,92 @@
+import { Injectable } from "@angular/core";
+import { ActivatedRoute, Params, Router } from "@angular/router";
+import { environment } from "../../environments/environment";
+import { TokenService, UserSessionService } from "@boklisten/bl-connect";
+import { Subject } from "rxjs";
+import { Observable } from "rxjs/internal/Observable";
+
+@Injectable({
+	providedIn: "root",
+})
+export class BlNextLinkerService {
+	private readonly _login$: Subject<boolean>;
+	private readonly _logout$: Subject<boolean>;
+	private tokensReceived = false;
+
+	constructor(
+		private _tokenService: TokenService,
+		private _userSessionService: UserSessionService,
+		private route: ActivatedRoute,
+		private router: Router,
+		private tokenService: TokenService
+	) {
+		this._login$ = new Subject<boolean>();
+		this._logout$ = new Subject<boolean>();
+		this.onBlConnectLogout();
+	}
+
+	public redirectToBlNext(subPath: string) {
```

Comment:
Not sure what code causes it, but if you try to log in with a non-employee user you get an infinite relog loop where it redirects back to next without some `unauthorized`-message. I think the best solution is to show an unauthorized-page with a button to log out/try another user.

Reply:
Added this to a Trello card. Not high priority for now, but should be fixed sooner rather than later.